x-pack/filebeat/input/s3/input_test.go

```diff
@@ -0,0 +1,247 @@
+		key:    "log2019-06-21-16-16-54",
+		region: "us-west-1",
+	}
+)
+
+func (m *MockS3Client) GetObjectRequest(input *s3.GetObjectInput) s3.GetObjectRequest {
+	logBody := ioutil.NopCloser(bytes.NewReader([]byte(s3LogString1 + s3LogString2)))
+	httpReq, _ := http.NewRequest("", "", nil)
+	return s3.GetObjectRequest{
+		Request: &awssdk.Request{
+			Data: &s3.GetObjectOutput{
+				Body: logBody,
+			},
+			HTTPRequest: httpReq,
+		},
+	}
+}
+
+func TestGetRegionFromQueueURL(t *testing.T) {
+	queueURL := "https://sqs.us-east-1.amazonaws.com/627959692251/test-s3-logs"
+	regionName, err := getRegionFromQueueURL(queueURL)
+	assert.NoError(t, err)
+	assert.Equal(t, "us-east-1", regionName)
+}
+
+func TestHandleMessage(t *testing.T) {
+	cases := []struct {
+		title           string
+		message         sqs.Message
+		expectedS3Infos []s3Info
+	}{
+		{
+			"sqs message with event source aws:s3 and event name ObjectCreated:Put",
+			sqs.Message{
+				Body: awssdk.String("{\"Records\":[{\"eventSource\":\"aws:s3\",\"awsRegion\":\"ap-southeast-1\",\"eventTime\":\"2019-06-21T16:16:54.629Z\",\"eventName\":\"ObjectCreated:Put\",\"s3\":{\"configurationId\":\"object-created-event\",\"bucket\":{\"name\":\"test-s3-ks-2\",\"arn\":\"arn:aws:s3:::test-s3-ks-2\"},\"object\":{\"key\":\"server-access-logging2019-06-21-16-16-54-E68E4316CEB285AA\"}}}]}"),
+			},
+			[]s3Info{
+				{
+					name: "test-s3-ks-2",
+					key:  "server-access-logging2019-06-21-16-16-54-E68E4316CEB285AA",
+				},
+			},
+		},
+		{
+			"sqs message with event source aws:s3 and event name ObjectCreated:Delete",
+			sqs.Message{
+				Body: awssdk.String("{\"Records\":[{\"eventSource\":\"aws:s3\",\"awsRegion\":\"ap-southeast-1\",\"eventTime\":\"2019-06-21T16:16:54.629Z\",\"eventName\":\"ObjectCreated:Delete\",\"s3\":{\"configurationId\":\"object-created-event\",\"bucket\":{\"name\":\"test-s3-ks-2\",\"arn\":\"arn:aws:s3:::test-s3-ks-2\"},\"object\":{\"key\":\"server-access-logging2019-06-21-16-16-54-E68E4316CEB285AA\"}}}]}"),
+			},
+			[]s3Info{},
+		},
+		{
+			"sqs message with event source aws:ec2 and event name ObjectCreated:Put",
+			sqs.Message{
+				Body: awssdk.String("{\"Records\":[{\"eventSource\":\"aws:ec2\",\"awsRegion\":\"ap-southeast-1\",\"eventTime\":\"2019-06-21T16:16:54.629Z\",\"eventName\":\"ObjectCreated:Put\",\"s3\":{\"configurationId\":\"object-created-event\",\"bucket\":{\"name\":\"test-s3-ks-2\",\"arn\":\"arn:aws:s3:::test-s3-ks-2\"},\"object\":{\"key\":\"server-access-logging2019-06-21-16-16-54-E68E4316CEB285AA\"}}}]}"),
+			},
+			[]s3Info{},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.title, func(t *testing.T) {
+			s3Info, err := handleSQSMessage(c.message)
+			assert.NoError(t, err)
+			assert.Equal(t, len(c.expectedS3Infos), len(s3Info))
+			if len(s3Info) > 0 {
+				assert.Equal(t, c.expectedS3Infos[0].key, s3Info[0].key)
+				assert.Equal(t, c.expectedS3Infos[0].name, s3Info[0].name)
+			}
+		})
+	}
+}
+
+func TestNewS3BucketReader(t *testing.T) {
+	p := Input{}
```

Comment:
nit: you don't need and Input at all? If all you need is an `context`, then only instantiate an context (or pass context.Background()).

Reply:
😬 not sure what I was thinking. Fixed. Thanks!